samples/snippets/create_test_resources.py

```diff
@@ -0,0 +1,132 @@
+        print(
+            "Created bucket {} in {} with storage class {}".format(
+                new_bucket.name, new_bucket.location, new_bucket.storage_class
+            )
+        )
+        return new_bucket
+
+
+def check_if_bucket_exists(new_bucket_name):
+    """Check if bucket is already exists"""
+    bucket_exists = False
+    buckets = storage_client.list_buckets()
+    for bucket in buckets:
+        if bucket.name == new_bucket_name:
+            bucket_exists = True
+            break
+    return bucket_exists
+
+
+def upload_data_to_bucket(bucket: Bucket):
+    """Upload data to a GCS bucket"""
+    blob = bucket.blob(object_name)
+    blob.upload_from_filename(resource_file)
+    print("Data from {} has being uploaded to {}".format(resource_file,
+                                                         bucket.name))
+
+
+def get_product_service_client():
+    """Get product service client"""
+    client_options = ClientOptions()
+    return ProductServiceClient(client_options=client_options)
+
+
+def get_import_products_gcs_request():
+    """Get import products from gcs request"""
+    gcs_bucket = "gs://{}".format(bucket_name)
+    gcs_errors_bucket = "{}/error".format(gcs_bucket)
+
+    gcs_source = GcsSource()
+    gcs_source.input_uris = ["{0}/{1}".format(gcs_bucket, object_name)]
+
+    input_config = ProductInputConfig()
+    input_config.gcs_source = gcs_source
+
+    errors_config = ImportErrorsConfig()
+    errors_config.gcs_prefix = gcs_errors_bucket
+
+    import_request = ImportProductsRequest()
+    import_request.parent = default_catalog
+    import_request.reconciliation_mode = ImportProductsRequest.ReconciliationMode.INCREMENTAL
+    import_request.input_config = input_config
+    import_request.errors_config = errors_config
+
+    print("---import products from google cloud source request---")
+    print(import_request)
+
+    return import_request
+
+
+def import_products_from_gcs():
+    """Call the Retail API to import products"""
+    import_gcs_request = get_import_products_gcs_request()
+    gcs_operation = get_product_service_client().import_products(
+        import_gcs_request)
+    print(
+        "Import operation is started: {}".format(gcs_operation.operation.name))
+
+    while not gcs_operation.done():
+        print("Please wait till operation is completed")
+        time.sleep(5)
+    print("Import products operation is completed")
+
+    if gcs_operation.metadata is not None:
+        print("Number of successfully imported products")
+        print(gcs_operation.metadata.success_count)
+        print("Number of failures during the importing")
+        print(gcs_operation.metadata.failure_count)
+    else:
+        print("Operation.metadata is empty")
+
+    print(
+        "Wait 2 -5 minutes till products become indexed in the catalog,\
+after that they will be available for search")
+
+
+upload_data_to_bucket(create_bucket(bucket_name))
```

Comment:
Can we pull `create_bucket` into a separate line? I'm also wondering if we should append a random suffix to the bucket to avoid issues when running tests concurrently.

As an example, please see https://github.com/googleapis/python-monitoring/blob/main/samples/snippets/v3/alerts-client/snippets_test.py#L37

Reply:
We are going to use the bucket as a long-living resource, we are storing it's name in the SecretManager. This code will be run in case if the data in the catalog is destroyed/removed just to recover the test data

Reply:
I'll merge this PR but I'd like to continue the discussion in #133 to understand if there is potential to setup/tear down resources as required.